### PR TITLE
Add koji tag and build root creator

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
proposed changes to nightly tags:

$ ./tools.rb  koji  ./configs/katello_nightly.yaml 


PROCESSING katello-thirdparty-candlepin-rhel7


PROCESSING katello-thirdparty-pulp-rhel7


PROCESSING katello-thirdparty-pulp-rhel6


PROCESSING katello-nightly-rhel7
[noop] kkoji remove-external-repo --tag=katello-nightly-rhel7-build jpackage-5.0-generic-free
[noop] kkoji remove-tag-inheritance katello-nightly-rhel7-build katello-nightly-rhel7-override
[noop] kkoji remove-tag-inheritance katello-nightly-rhel7-build foreman-plugins-nightly-rhel7-override
[noop] kkoji remove-tag-inheritance katello-nightly-rhel7-build foreman-nightly-nonscl-rhel7
[noop] kkoji remove-tag-inheritance katello-nightly-rhel7-build katello-thirdparty-pulp-rhel7
[noop] kkoji remove-tag-inheritance katello-nightly-rhel7-build foreman-nightly-rhel7
[noop] kkoji add-tag-inheritance --priority=0 katello-nightly-rhel7-build katello-nightly-rhel7-override
[noop] kkoji add-tag-inheritance --priority=3 katello-nightly-rhel7-build foreman-plugins-nightly-rhel7-override
[noop] kkoji add-tag-inheritance --priority=4 katello-nightly-rhel7-build foreman-nightly-nonscl-rhel7
[noop] kkoji add-tag-inheritance --priority=10 katello-nightly-rhel7-build foreman-nightly-rhel7
[noop] kkoji regen-repo katello-nightly-rhel7-build


PROCESSING katello-nightly-rhel6
[noop] kkoji remove-external-repo --tag=katello-nightly-rhel6-build qpid-copr-again
[noop] kkoji remove-external-repo --tag=katello-nightly-rhel6-build jpackage-5.0-generic-free
[noop] kkoji remove-external-repo --tag=katello-nightly-rhel6-build centos-sclo-rh-rhel-6
[noop] kkoji remove-external-repo --tag=katello-nightly-rhel6-build centos-sclo-sclo-rhel-6
[noop] kkoji remove-tag-inheritance katello-nightly-rhel6-build katello-nightly-rhel6-override
[noop] kkoji remove-tag-inheritance katello-nightly-rhel6-build foreman-plugins-nightly-rhel6-override
[noop] kkoji remove-tag-inheritance katello-nightly-rhel6-build foreman-nightly-nonscl-rhel6
[noop] kkoji remove-tag-inheritance katello-nightly-rhel6-build katello-thirdparty-pulp-rhel6
[noop] kkoji remove-tag-inheritance katello-nightly-rhel6-build foreman-nightly-rhel6
[noop] kkoji add-tag-inheritance --priority=0 katello-nightly-rhel6-build katello-nightly-rhel6-override
[noop] kkoji regen-repo katello-nightly-rhel6-build


PROCESSING katello-nightly-rhel5
[noop] kkoji regen-repo katello-nightly-rhel5-build


PROCESSING katello-nightly-fedora23
[noop] kkoji remove-tag-inheritance katello-client-fedora23 katello-thirdparty-pulp-fedora23
[noop] kkoji remove-tag-inheritance katello-client-fedora23 katello-nightly-fedora23
[noop] kkoji add-tag-inheritance --priority=0 katello-client-fedora23 katello-nightly-fedora23
[noop] kkoji add-tag-inheritance --priority=2 katello-client-fedora23 katello-thirdparty-pulp-fedora23
[noop] kkoji regen-repo katello-nightly-fedora23-build


PROCESSING katello-nightly-fedora24
[noop] kkoji add-tag-inheritance --priority=0 katello-nightly-fedora24-build katello-nightly-fedora24
[noop] kkoji remove-tag-inheritance katello-client-fedora24 katello-nightly-fedora24
[noop] kkoji add-tag-inheritance --priority=0 katello-client-fedora24 katello-nightly-fedora24
[noop] kkoji add-tag-inheritance --priority=2 katello-client-fedora24 katello-thirdparty-pulp-fedora24
[noop] kkoji regen-repo katello-nightly-fedora24-build


PROCESSING katello-nightly-fedora25
[noop] kkoji add-tag katello-nightly-fedora25 --arches=x86_64
[noop] kkoji add-tag katello-nightly-fedora25-build --arches=x86_64
[noop] kkoji add-target katello-nightly-fedora25 katello-nightly-fedora25-build
[noop] kkoji add-group katello-nightly-fedora25-build build
[noop] kkoji add-group-pkg katello-nightly-fedora25-build build bash,bzip2,coreutils,cpio,diffutils,fedora-release,findutils,gawk,gcc,gcc-c++,grep,gzip,info,make,patch,redhat-release-server,redhat-rpm-config,rpm-build,ruby193-build,scl-utils-build,sed,shadow-utils,tar,unzip,util-linux,which,xz
[noop] kkoji add-group katello-nightly-fedora25-build srpm-build
[noop] kkoji add-group-pkg katello-nightly-fedora25-build srpm-build bash,curl,fedora-release,gnupg2,make,redhat-release-server,redhat-rpm-config,rpm-build,ruby193-build,scl-utils-build,shadow-utils
[noop] kkoji add-external-repo --tag=katello-nightly-fedora25-build fedora-25-updates
[noop] kkoji add-external-repo --tag=katello-nightly-fedora25-build fedora-25
[noop] kkoji add-tag katello-client-fedora25 --arches=x86_64
[noop] kkoji add-tag-inheritance --priority=0 katello-nightly-fedora25-build katello-nightly-fedora25
[noop] kkoji add-tag-inheritance --priority=0 katello-client-fedora25 katello-nightly-fedora25
[noop] kkoji regen-repo katello-nightly-fedora25-build

File: configs/katello_33.yaml [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    :version_branch: true
+  :katello-selinux:
+    :branch: KATELLO-3.3
+    :repo: https://github.com/Katello/katello-selinux.git
+  :katello-installer:
+    :branch: KATELLO-3.3
+    :repo: https://github.com/Katello/katello-installer.git
+    :version_branch: true
+  :hammer-cli-katello:
+    :branch: 0.1.0
```
Comment (ehelms) [inline configs/katello_33.yaml]: We probably would want this updated for 3.3?

File: configs/katello_33.yaml [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    :repo: https://github.com/Katello/hammer-cli-katello.git
+  :katello.org:
```
Comment (ehelms) [inline configs/katello_33.yaml]: No longer a thing!

File: configs/katello_33.yaml [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+  :3.3:
+    :redmine_release_id: 188
+:code_name: Baltic Porter
+:repos:
+  :katello:
+    :main: true
+    :branch: KATELLO-3.3
+    :repo: https://github.com/Katello/katello.git
+    :version_branch: true
+  :katello-selinux:
```
Comment (ehelms) [inline configs/katello_33.yaml]: Were there updates to this? I don't know if we have been branching it given it doesn't get updates often.

File: configs/katello_33.yaml [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    :branch: 5.1.1
+    :repo: https://github.com/Katello/puppet-katello.git
+  :puppet-capsule:
+    :branch: 3.1.2
+    :repo: https://github.com/Katello/puppet-capsule.git
+  :puppet-pulp:
+    :branch: 3.4.0
+    :repo: https://github.com/Katello/puppet-pulp.git
+  :katello-agent:
+    :branch: 2.6.0
```
Comment (ehelms) [inline configs/katello_33.yaml]: This is up to 2.7.0. Looking at the repo, we should release a new one with the tracer support.

File: configs/katello_33.yaml [hunk 5]
```diff
@@ -0,0 +1,178 @@
+---
+:project: katello
```
Comment (ehelms) [inline configs/katello_33.yaml]: I wish I'd never used symbols as the config entry (used straight strings).

File: configs/katello_33.yaml [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+      :katello-3.3-fedora25: {}
+      :katello-3.3-client-fedora25:
+        :katello-3.3-fedora25: 0
+    :external_repos:
+      - fedora-25-updates
+      - fedora-25
+
+
+
+
```
Comment (ehelms) [inline configs/katello_33.yaml]: So many lines!

File: configs/katello_nightly.yaml [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+      :katello-nightly-rhel5-override:
+        :katello-nightly-rhel5: 0
+      :katello-nightly-rhel5: {}
+      :katello-client-rhel5:
+        :katello-nightly-rhel5: 0
+        :katello-thirdparty-pulp-rhel5: 2
+    :external_repos:
+      - rhel-5.9-server
+      - epel-5
+  - :name: katello-nightly-fedora23
```
Comment (ehelms) [inline configs/katello_nightly.yaml]: We should probably drop this for nightly?

File: lib/tool_belt/koji_tools.rb [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+
+    def regen_repo(tag)
+      make_change("regen-repo #{tag}")
+    end
+
+    def tag_exists?(tag_name)
+      koji_info("taginfo #{tag_name}")[1]
+    end
+
+    def is_build_target?(name, build_target)
```
Comment (ehelms) [inline lib/tool_belt/koji_tools.rb]: Could drop the `is` here as the `?` takes care of that.

File: lib/tool_belt/koji_tools.rb [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+
+    def remove_inheritance(root, branch)
+      make_change("remove-tag-inheritance #{root} #{branch}")
+    end
+
+    def koji_info(command)
+      @systools.execute("#{KojiBuilder::KOJI_COMMAND} #{command}", true)
+    end
+
+    def make_change(command)
```
Comment (ehelms) [inline lib/tool_belt/koji_tools.rb]: What happens if I give this `$1` ?

File: configs/katello_33.yaml (hunk 3 above)
Comment (jlsherrill) [inline configs/katello_33.yaml]: don't think so, i just copied all this from 3.2's config file.  I don't think its needed (but i don't think some below it are either).

File: lib/tool_belt/koji_tools.rb (hunk 9 above)
Comment (jlsherrill) [inline lib/tool_belt/koji_tools.rb]: I don't quite understand what you mean

File: configs/katello_33.yaml (hunk 6 above)
Comment (jlsherrill) [inline configs/katello_33.yaml]: you caught me, i've been embedding programs in my [white space](https://en.wikipedia.org/wiki/Whitespace_(programming_language))

File: configs/katello_33.yaml (hunk 5 above)
Comment (jlsherrill) [inline configs/katello_33.yaml]: updated my new parts to use strings

File: configs/katello_33.yaml (hunk 3 above)
Comment (jlsherrill) [inline configs/katello_33.yaml]: sorry, i see what you mean now.  yes there were (from months ago).  I've updated that and started a new build in katello-packaging

File: configs/katello_33.yaml (hunk 1 above)
Comment (jlsherrill) [inline configs/katello_33.yaml]: released and updated

File: configs/katello_33.yaml (hunk 4 above)
Comment (jlsherrill) [inline configs/katello_33.yaml]: bumped to 2.7.0

File: configs/katello_nightly.yaml (hunk 7 above)
Comment (jlsherrill) [inline configs/katello_nightly.yaml]: dropped

File: lib/tool_belt/koji_tools.rb (hunk 8 above)
Comment (jlsherrill) [inline lib/tool_belt/koji_tools.rb]: done

**Comment by jlsherrill** [comment]
added  a .travis.yaml

any thoughts about moving this repo to the katello org?

File: lib/tool_belt/koji_tools.rb (hunk 9 above)
Comment (ehelms) [inline lib/tool_belt/koji_tools.rb]: ....will it give me 4 quarters? or 10 dimes?

File: configs/katello_33.yaml [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+    :version_branch: true
+  :hammer-cli-katello:
+    :branch: 0.3.0
+    :repo: https://github.com/Katello/hammer-cli-katello.git
+  :katello-packaging:
+    :branch: KATELLO-3.3
+    :repo: https://github.com/Katello/katello-packaging.git
+    :version_branch: true
+  :katello-agent:
+    :branch: 2.7.0
```
Comment (ehelms) [inline configs/katello_33.yaml]: 2.8.0 ?

**Comment by jlsherrill** [comment]
@ehelms ping,  good with this?

**Comment by ehelms** [comment]
Mind re-pushing to this? I want to see if the travis took effect.

I'd be OK moving this to either organization. I suppose we use it the heaviest currently so moving to Katello would be best for now.

**Comment by jlsherrill** [comment]
had to add 'rake' to the Gemfile, but its all green now

**Pull request merged**